src/main/resources/application.yml

```diff
@@ -14,6 +14,8 @@ spring:
       WRITE_DATES_AS_TIMESTAMPS: false
 
   security:
+    filter:
+      dispatcher-types: request,error,forward,include
```

Comment:
Spring Security 5.x used to just do `request` and `error` dispatch types, so reading https://docs.spring.io/spring-security/reference/5.8/migration/servlet/authorization.html had to exclude `async` here.  Regarding `async` requests there appear to be two calls, with the second one then using an anonymous context (think it is because it happens on a separate thread from the first one).  https://stackoverflow.com/questions/37138529/does-the-spring-security-filterchain-always-run-twice-for-asynchronous-requests suggests that it happens.  This worked in previous versions of spring security as `async` wasn't then secured.

Reply:
I don't really understand what the async dispatcher type call is and what we should have done differently to make it secured. Luckily these tests are only used in the dev environment so I think this workaround is fine.